cmakebuild.js

```diff
@@ -6,9 +6,12 @@ function runCmake (opts, target, cb) {
     if (err) return cb(err)
 
     var args = ['rebuild']
-    args.push('--runtime-version=' + target)
+    if (opts.runtime !== 'napi') args.push('--runtime-version=' + target)
     args.push('--arch=' + opts.arch)
-    args.push('--runtime=' + opts.runtime)
+    if (opts.runtime !== 'napi') args.push('--runtime=' + opts.runtime)
+    if (opts.runtime === 'napi' && target) {
+      args.push('--CDnapi_build_version=' + target)
```

Comment:
What happens when `target` isn't defined?

Reply:
If `target` is not defined, the `napi_build_version` CMake customizable setting is not specified to CMake. In this case the `NAPI_VERSION` C preprocessor define value created by CMake is empty and the N-API build defaults to the most current N-API version supported by the Node version being built against.

I've not been able to create a situation where this happens. But I imagine the `prebuild` project could possibly evolve in such a way that it might happen. If so, the code is ready.